ui/src/dataExplorer/components/DataExplorer.tsx

```diff
@@ -9,40 +9,46 @@ import RateLimitAlert from 'src/cloud/components/RateLimitAlert'
 
 // Actions
 import {setActiveTimeMachine} from 'src/timeMachine/actions'
+import {setBuilderBucketIfExists} from 'src/timeMachine/actions/queryBuilder'
 
 // Utils
 import {HoverTimeProvider} from 'src/dashboards/utils/hoverTime'
 import {queryBuilderFetcher} from 'src/timeMachine/apis/QueryBuilderFetcher'
+import {readQueryParams} from 'src/shared/utils/queryParams'
 
 interface DispatchProps {
   onSetActiveTimeMachine: typeof setActiveTimeMachine
+  onSetBuilderBucketIfExists: typeof setBuilderBucketIfExists
 }
 
 type Props = DispatchProps
-class DataExplorer extends PureComponent<Props, {}> {
-  constructor(props: Props) {
-    super(props)
 
-    props.onSetActiveTimeMachine('de')
+const DataExplorer: FC<Props> = ({
```

Comment:
just want to call out that switching from a `PureComponent` to a `FunctionComponent` *might* have performance implications since a `FC` always renders, while a `PureComponent` sometimes doesn't re-render. I don't have enough context to provide anything more helpful than that :|

Reply:
thanks.. I'll test it a bit..